hamt/hamt.go

```diff
@@ -471,20 +480,22 @@ func makeAsyncTrieGetLinks(dagService ipld.DAGService, onShardValue func(*shardV
 	}
 }
 
-func (ds *Shard) walkTrie(ctx context.Context, cb func(*shardValue) error) error {
+func (ds *Shard) walkTrie(ctx context.Context, cb func(*Shard) error) error {
```

Comment:
I'll raise a separate issue about this but we definitely need to unify this function with `getValue`.